packages/react-data-grid/src/Header.tsx

```diff
@@ -0,0 +1,185 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import classNames from 'classnames';
+
+import { resizeColumn } from './ColumnMetrics';
+import { getColumn } from './ColumnUtils';
+import HeaderRow, { HeaderRowProps } from './HeaderRow';
+import getScrollbarSize from './getScrollbarSize';
+import { HeaderRowType } from './common/enums';
+import { Column, ColumnMetrics, CellMetaData } from './common/types';
+
+interface HeaderRowData extends Pick<HeaderRowProps, 'filterable' | 'onFilterChange'> {
+  height?: number;
+  rowType: HeaderRowType;
+}
+
+type SharedHeaderRowProps = Pick<HeaderRowProps,
+'draggableHeaderCell'
+| 'getValidFilterValues'
+| 'sortDirection'
+| 'sortColumn'
+| 'onHeaderDrop'
+| 'onSort'
+| 'onScroll'
+>;
+
+export interface Props extends SharedHeaderRowProps {
+  columnMetrics: ColumnMetrics;
+  totalWidth?: number;
+  height: number;
+  headerRows: HeaderRowData[];
+  onColumnResize?(pos: number, width: number): void;
```

Comment:
This can be changed to a required prop
https://github.com/adazzle/react-data-grid/blob/next/packages/react-data-grid/src/ReactDataGrid.js#L805

Reply:
I think same for `totalWidth`
https://github.com/adazzle/react-data-grid/blob/next/packages/react-data-grid/src/ReactDataGrid.js#L802